src/base/kaldi-types.h

```diff
@@ -38,14 +38,26 @@ typedef float   BaseFloat;
 // we can do this a different way if some platform
 // we find in the future lacks stdint.h
 #include <stdint.h>
+#if _MSC_VER >= 1900
+#include <fst/types.h>
+#endif  // _MSCVER >= 1900
 
 namespace kaldi {
-typedef uint16_t        uint16;
-typedef uint32_t        uint32;
-typedef uint64_t        uint64;
-typedef int16_t         int16;
-typedef int32_t         int32;
-typedef int64_t         int64;
+#if _MSC_VER >= 1900
```

Comment:
Maybe this should not even be conditional on MSC? The types would be correct either way, if my understanding is correct.